src/lean_lsp_mcp/server.py

```diff
@@ -323,21 +323,48 @@ def completions(
     rel_path = setup_client_for_file(ctx, file_path)
     if not rel_path:
         return "No valid lean file path found. Could not set up client and load file."
-    update_file(ctx, rel_path)
+    content = update_file(ctx, rel_path)
 
     client: LeanLSPClient = ctx.request_context.lifespan_context.client
     completions = client.get_completions(rel_path, line - 1, column - 1)
 
+    lines = content.splitlines()
+    if line > 0 and line <= len(lines):
```

Comment:
this part of the code is crap and might benefit from some rewriting. we're just trying to extract the previous words here.